deduce.go

```diff
@@ -690,6 +695,253 @@ func (sm *SourceMgr) deduceFromPath(path string) (deductionFuture, error) {
 	}, nil
 }
 
+type deductionCoordinator struct {
+	ctx      context.Context
```

Comment:
Yeah, so, I realize I actually conflated two things. *THAT* stored context doesn't necessarily need to be there; it could be grabbed from the `callManager` on the fly in the actor loop. But that doesn't really change much, as there's still a context on the `callManager` struct.

The basic issue here comes down to there being two different possible origins for cancellation of any given `SourceManager` method call:

1. A cancellation of the context passed directly into the method - the standard Go way of doing it
2. A cancellation from the long-lived context (e.g., a signal is sent to a `dep` process, which then needs to tell the `sourceMgr` to shut down via a context that's injected in the `NewSourceManager()` call), which must propagate to _all_ open calls

There are different possible approaches to this, but that's the essential reality, and dealing with it entails combining cancelation paths somehow.